apps/OctoWatts/octowatts.star

```diff
@@ -0,0 +1,480 @@
+            demand_value = -abs(demand_value)
+            demand_average = -abs(demand_average)
+
+    # Calculate percentages
+    percentage = calculate_percentage(demand_value, RANGEMIN, RANGEMAX)
+    trend_percentage = calculate_trend_percentage(demand_value, demand_average)
+
+    # Draw trend arrow and usage text
+    trend = draw_arrow(demand_value, demand_average)
+    demand_text = format_demand_text(demand_value)
+
+    # Get colours and weights
+    trend_color = get_color(int(trend_percentage))
+    color = get_color(int(percentage))
+
+    # Render root content
+    if CHARTTYPE == "bar_horizontal":
+        pixels, pixels_border = calculate_pixels(CHARTTYPE, percentage, demand_value)
+        return render_content("bar_horizontal", pixels, pixels_border, demand_text, trend, trend_color, color, None, None, None, None)
+    elif CHARTTYPE == "bar_vertical":
+        pixels, pixels_border = calculate_pixels(CHARTTYPE, percentage, demand_value)
+        return render_content("bar_vertical", pixels, pixels_border, demand_text, trend, trend_color, color, None, None, None, None)
+    else:
+        weights = generate_weights(percentage)
+        return render_content("pie", None, None, demand_text, trend, trend_color, color, weights, None, None, None)
+
+def basic_random():
+    seed = str(time.now())[0:27].replace(":", "").replace(".", "").replace("-", "").replace(" ", "").replace("+", "")
+    seed = int(seed)
+    return (seed * 48271) % 2147483647
+
+def random_in_range(min_val, max_val):
+    # Generate a pseudo-random number
+    rand_value = basic_random()
+
+    # Scale to the desired range
+    range_size = max_val - min_val + 1
+    return min_val + (rand_value % range_size)
+
+def process_telemetry(telemetry):
+    # Extract the latest demand value and compare with the previous one
+    latest_telemetry = telemetry[-1]
+
+    # Check if there are at least 6 telemetry entries
+    if len(telemetry) >= 7:
+        # Extract previous telemetry data from -6 to -2 (for the last minute)
+        previous_telemetries = telemetry[-7:-1]  # Get entries from -6 to -2 (non-inclusive of -2)
+        print(previous_telemetries)
+
+        # Initialize total and count for averaging
+        total_demand = 0.0
+        count = 0
+
+        # Calculate the total of previous demand values
+        for t in previous_telemetries:
+            demand = float(t.get("demand", 0))
+            total_demand += demand
+            count += 1
+
+        # Calculate average
+        demand_average = total_demand / count if count > 0 else 0
+
+    else:
+        # Not enough entries; set to current demand
+        demand_average = latest_telemetry["demand"]
+
+    demand_value = int(float(latest_telemetry.get("demand", 0)))
+    return demand_value, demand_average
+
+def fetch_telemetry(access_token, device_id, ttl):
+    # Check cache so we don't make requests more than every 10 seconds
+    cached_telemetry = cache.get("telemetry".format(device_id))
```

Comment:
Wouldn't this just evaluate to "telemetry"? No `{}`in that string. That in turn means the same cache key is used for all installations of the app.

Reply:
Your right, I've pushed the update code.